lib/strategy.js

```diff
@@ -254,46 +248,46 @@ Strategy.prototype._loadUserProfile = function(accessToken, params, done) {
 
 /**
 * Retrieve user profile from Yahoo.
-* inpired from post: http://yahoodevelopers.tumblr.com/post/105969451213/implementing-yahoo-oauth2-authentication
-* other code from : passport-yahoo-token repo
 * This function constructs a normalized profile, with the following properties:
 *
 *   - `id`
 *   - `displayName`
+*   - `name`
 *
 * @param {String} token
 * @param {String} tokenSecret
 * @param {Object} params
 * @param {Function} done
 * @api protected
 */
-Strategy.prototype.userProfile = function (accessToken, params, done) {
-  this._oauth2._useAuthorizationHeaderForGET = true;
-  var request_options = {
-    url: this._options.profileURL.replace(':xoauthYahooGuid', params.xoauth_yahoo_guid),
-    headers: {
-      Authorization: this._oauth2.buildAuthHeader(accessToken)
-    },
-    rejectUnauthorized: false,
-    json: true
-  };
-
-  request.get(request_options, function(err, response, json) {
+Strategy.prototype.userProfile = function (accessToken, done) {
+  this._oauth2.useAuthorizationHeaderforGET(true);
+
+  this._oauth2.get(this._options.profileURL, accessToken, function(err, body, res) {
     if(err) {
       return done(new InternalOAuthError('Failed to fetch user profile', err));
     }
-    if (response.statusCode !== 200) {
-      return done(new InternalOAuthError('Failed to fetch user profile', json));
+    if(res.statusCode !== 200) {
+      return done(new InternalOAuthError('Failed to fetch user profile', body));
     }
-    try{
 
-      var profile = { provider: 'yahoo' };
-      profile.id = json.profile.guid;
-      profile.displayName = json.profile.givenName + ' ' + json.profile.familyName;
-      profile.name = { familyName: json.profile.familyName,
-                       givenName: json.profile.givenName };
+    let json;
+    try {
+      json = JSON.parse(body);
+    } catch(e) {
+      return done(new Error('Failed to parse user profile'));
+    }
 
-      profile._raw = json;
+    try{
+      const profile = { provider: 'yahoo' };
+      profile.id = json.sub;
```

Comment:
Sub is exactly the same ID?

Reply:
It would appear so according to https://developer.yahoo.com/oauth/social-directory-eol/ but I will test it before merging.